packages/taiko-client/prover/proof_submitter/proof_submitter.go

```diff
@@ -159,9 +162,9 @@ func (s *ProofSubmitter) RequestProof(ctx context.Context, meta metadata.TaikoBl
 				log.Error("Failed to request proof, context is canceled", "blockID", opts.BlockID, "error", ctx.Err())
 				return nil
 			}
-			// Check if the proof buffer is full
-			if s.proofBuffer.MaxLength > 1 && s.proofBuffer.MaxLength == uint64(s.proofBuffer.Len()) {
-				log.Debug("Buffer is full now", "blockID", meta.GetBlockID())
+			// Check if the proof buffer is full.
+			if s.proofBuffer.Enabled() && uint64(s.proofBuffer.Len()) >= s.proofBuffer.MaxLength {
```

Comment:
```suggestion
			if s.proofBuffer.Enabled() && uint64(s.proofBuffer.Len()) > s.proofBuffer.MaxLength {
```

Reply:
If `uint64(s.proofBuffer.Len())` is equal to `s.proofBuffer.MaxLength`, its also full.